### ISSUE TITLE
Compiling issues when including in larger package

### ISSUE THREAD

**Issue by mankoff** [nbren12/call_py_fort#25]
I'm getting the following errors when I try to compile this package:

```

   14 |     builtins.print(STATE)
      |     1
Error: Unclassifiable statement at (1)
LI_CPL.F90:13:26:

   13 |     call set_state("a", a)
      |                          1
Error: There is no specific subroutine for the generic ‘set_state’ at (1)
LI_CPL.F90:15:26:

   15 |     call get_state("a", a)
      |                          1
Error: There is no specific subroutine for the generic ‘get_state’ at (1)
```

I'm wondering if these error messages make more sense to you than I and if you have suggestions how to properly compile this.

In more detail...

I have been able to get `call_py_fort` working with a simple Fortran and Python program. I've compiled `call_py_fort` into `opt` and then can compile my demo program with:

```bash
export PYTHONPATH=.:$(pwd)/examples:$PYTHONPATH

cpf_path=/home/kdm/projects/GISS/call_py_fort
export LD_LIBRARY_PATH=${cpf_path}/opt/lib:/home/kdm/local/mambaforge/envs/fort2py/lib

gfortran \
  ex1.f90\
  -I${cpf_path}/opt/include\
  -Wl,-rpath=${cpf_path}/opt/lib\
  -L${cpf_path}/opt/lib \
  -lcallpy \

./a.out
```

I'm having trouble replicating this for a much larger package with its own historical custom build system.

I've created a small module rather than a standalone binary:

```fortran
module LI_CPL_mod
  
  use callpy_mod
  
  implicit none
  integer a
  
contains

  subroutine LI_CPL
    a = 42
    call set_state("a", a)
    builtins.print(STATE)
    call get_state("a", a)
    WRITE(*,*) "AAA", a
  end subroutine LI_CPL

end module LI_CPL_mod
```

And elsewhere `call LI_CPL`.

When I try to compile, I get these errors:

```bash
gfortran -c -o LI_CPL.o -g -cpp -fconvert=big-endian -O2 -fno-range-check -ffree-line-length-none  -I/home/kdm/projects/GISS/LIME/opt/include -I/home/kdm/projects/GISS/modelE_C/model/mod -I${cpf_path}/opt/include -O0 -ggdb3 -fwrapv  -fallow-argument-mismatch -fallow-invalid-boz -DLI_COUPLE_DAILY -pg -Wl,-rpath=/home/kdm/projects/GISS/call_py_fort/opt/lib -DMACHINE_Linux -DUSE_MPI -DMPITYPE_LOOKUP_HACK -I/home/kdm/projects/GISS/modelE_C/model/shared -I/home/kdm/projects/GISS/modelE_C/model/include -L${cpf_path}/opt/lib -lcallpy  LI_CPL.F90

LI_CPL.F90:14:5:

   14 |     builtins.print(STATE)
      |     1
Error: Unclassifiable statement at (1)
LI_CPL.F90:13:26:

   13 |     call set_state("a", a)
      |                          1
Error: There is no specific subroutine for the generic ‘set_state’ at (1)
LI_CPL.F90:15:26:

   15 |     call get_state("a", a)
      |                          1
Error: There is no specific subroutine for the generic ‘get_state’ at (1)
```

**Comment by mankoff** [comment]
I got it compiling but now see 

```
function my_plugin.call_function() called, but initialization code failed.  Returning 0.
function my_plugin.get_state_py() called, but initialization code failed.  Returning 0.
function my_plugin.set_state_py() called, but initialization code failed.  Returning 0.
function my_plugin.call_function() called, but initialization code failed.  Returning 0.
function my_plugin.get_state_py() called, but initialization code failed.  Returning 0.
```

**Comment by mankoff** [comment]
Solved with the help of #12. Re-installed mamba environment and it worked. Conda/mamba is just:

```
mamba create -n fort2py
mamba activate fort2py
mamba install -y cffi numpy
```

And then point there with appropriate INCLUDE and LIBRARY and PYTHONPATH variables.
